commit d8530be8b5fdc6f65bd11beb8a12e0819eee6575
Author: Tobias Doerffel <tobias.doerffel@gmail.com>
Date:   Wed May 8 00:18:44 2013 +0200

    ItalcCore/ItalcVncConnection: exit main loop earlier if requested
    
    If m_stopped is set after waiting for a message, do not finish the
    main loop iteration but exit immediately.

diff --git a/lib/src/ItalcVncConnection.cpp b/lib/src/ItalcVncConnection.cpp
index 0e08f80..e147892 100644
--- italc-2.0.0.ORIG/lib/src/ItalcVncConnection.cpp
+++ italc-2.0.0/lib/src/ItalcVncConnection.cpp
@@ -618,7 +618,7 @@ void ItalcVncConnection::doConnection()
 			timeout = 100*1000;	// 100 ms
 		}
 		const int i = WaitForMessage( m_cl, timeout );
-		if( i < 0 )
+		if( m_stopped || i < 0 )
 		{
 			break;
 		}
